.github/workflows/test.yml

```diff
@@ -104,31 +87,21 @@ jobs:
           echo "REQUESTS_HTTP_PROXY_AUTH_USER=test" >> $GITHUB_ENV
           echo "REQUESTS_HTTP_PROXY_AUTH_PASS=pass" >> $GITHUB_ENV
 
-      - name: Ensure the HTTPS test instance on Heroku is spun up
-        run: curl -s -I http://requests-php-tests.herokuapp.com/ > /dev/null
-
-      - name: Check mitmproxy version
-        run: mitmdump --version
-
-      - name: Ping localhost domain
-        run: ping -c1 localhost
-
-      - name: Access localhost on port 8080
-        run: curl -i http://localhost:8080
-
-      - name: Access localhost on port 9002
-        run: curl -i http://localhost:9002
-
-      - name: Show PHPUnit version
-        run: vendor/bin/phpunit --version
+      - name: Checkproxy server
+        run: |
+          mitmdump --version
+          curl -s -I http://requests-php-tests.herokuapp.com/ > /dev/null
+          ping -c1 localhost
+          curl -i http://localhost:8080
+          curl -i http://localhost:9002
 
       - name: Run the unit tests, no code coverage
-        if: ${{ matrix.coverage == false }}
+        if: ${{ ! matrix.coverage }}
         run: composer test
 
       - name: Run the unit tests with code coverage
-        if: ${{ matrix.coverage == true }}
-        run: vendor/bin/phpunit --coverage-clover clover.xml
+        if: ${{ matrix.coverage }}
+        run: composer exec -- phpunit --coverage-clover clover.xml
```

Comment:
Probably also groundbreaking: the `composer.json` for the project also has an explicit `coverage` script, so this could be changed to `composer coverage -- --coverage-closer clover.xml`.

Still, the overhead of running this via Composer is not worth it.